tensorflow-frameworks/tensorflow-data/README.md

```diff
@@ -0,0 +1,220 @@
+
+NOTE: This readme follows the discussion of this [RFC]()
+
+
+Tensorflow-Data (Java)
+==
+
+TensorFlow Data provides utilities and APIsfor loading data of various formats
+, and preparing datasets for use in training and using deep learning models
+. This package
+ provides a
+ simple API for configuring and iterating over
+ datasets in both "graph" and "eager" mode.
+
+Usage
+--
+
+The `Dataset` abstraction represents a sequence of elements, where each element in the sequence is a collection (`List`) of tensors (or, "components").
+
+
+
+The `Dataset` class represents a sequence of elements which can be iterated over and
+transformed. Each element is a list of "output" operands, represented by the type `List<Output<?>>`. 
+
+Note: An `Output` is a symbolic handle to a tensor produced by a TensorFlow op. In graph
+mode, `Output` objects will not have a concrete `Tensor` value unless all dependent operations
+are run in a `Session` (this is done "in-real-time" in eager mode).
+
+### Construction
+
+Datasets can be constructed either directly from a data source (e.g. a list of tensors representing the components of the dataset), or as a transformation on an existing dataset.
+
+#### From Data Source
+
+To construct a dataset from a list of tensor components, use 
+`Dataset.fromTensorSlices( ... )`. For example, say we are working
+with a standard feature/label dataset which has 4 elements.
+
+```java
+FloatNdArray features = StdArrays.ndCopyOf(
+        new float[][] {
+        {1, 2, 3},
+        {4, 5, 6},
+        {7, 8, 9},
+        {10, 11, 12}
+});
+
+FloatNdArray labels = StdArrays.ndCopyOf(
+    new float[] {
+        0,
+        1,
+        1,
+        0
+});
```

Comment:
In case of simple vectors, you probably want to go with `NdArrays.vectorOf(0, 1, 1, 0);`

Reply:
Makes sense!